models/rarible/polygon/rarible_polygon_events.sql

```diff
@@ -69,6 +69,70 @@ WITH trades AS (
         {% if is_incremental() %}
         AND evt_block_time >= date_trunc("day", now() - interval '1 week')
         {% endif %}
+
+    UNION ALL
+
+    -- directPurchase
+    SELECT 'buy' AS trade_category,
+        block_time AS evt_block_time,
+        block_number AS evt_block_number,
+        `hash` AS evt_tx_hash,
+        `to` AS contract_address,
+        CAST(-1 as integer) AS evt_index,
+        'Trade' AS evt_type,
+        `from` AS buyer,
+        '0x' || right(substring(`data`, 11 + 64, 64), 40) AS seller,
+        '0x' || right(substring(`data`, 11 + 64 * 17, 64), 40) AS nft_contract_address,
+        CAST(bytea2numeric_v3(substr(`data`, 11 + 64 * 18, 64)) AS string) AS token_id,
+        CAST(bytea2numeric_v3(substring(`data`, 11 + 64 * 14, 64)) AS DOUBLE) AS number_of_items,
+        CASE WHEN substring(`data`, 11 + 64 * 3, 8) = '73ad2146' THEN 'erc721' ELSE 'erc1155' END AS token_standard, -- 0x73ad2146: erc721; 0x973bb640: erc1155
+        CASE WHEN substring(`data`, 11 + 64 * 6, 64) = '0000000000000000000000000000000000000000000000000000000000000000'
+            THEN '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270'
+            ELSE '0x' || right(substring(`data`, 11 + 64 * 6, 64), 40)
+        END AS currency_contract,
+        CAST(bytea2numeric_v3(substring(`data`, 11 + 64 * 13, 64)) AS DOUBLE) AS amount_raw
+    FROM {{ source('polygon', 'transactions') }}
+    WHERE `to` = '0x12b3897a36fdb436dde2788c06eff0ffd997066e'
+        AND substring(`data`, 1, 10) = '0x0d5f7d35' -- 0x0d5f7d35: directPurchase; 0x67d49a3b: directAcceptBid
```

Comment:
please submit the contract (0x12b3897a36fdb436dde2788c06eff0ffd997066e) for decoding and make use of the decoded call tables instead of the transaction traces.